### PR TITLE
Fix Golang TestNttDeviceAsync

### PR DESCRIPTION
**Pull request by LeonHibnik**
## Describe the changes

This PR fixes TestNttDeviceAsync by adding a missing call to initDomain

## Linked Issues

Resolves #

### CODE REVIEW

**Review by ChickenLover — APPROVED** [review]

File: wrappers/golang/curves/bls12377/ntt_test.go [hunk 1]
```diff
@@ -120,6 +120,7 @@ func TestECNtt(t *testing.T) {
 
 func TestNttDeviceAsync(t *testing.T) {
 	cfg := GetDefaultNttConfig()
+	initDomain(largestTestSize, cfg)
```
Comment (jeremyfelder) [inline wrappers/golang/curves/bls12377/ntt_test.go:123]: Was this test consistently failing when running the entire test file or only when running this test?

Comment (LeonHibnik) [reply wrappers/golang/curves/bls12377/ntt_test.go:123]: Consistently on running only this test, randomly on CI/running all the tests

**Review by jeremyfelder — COMMENTED** [review]
I think it might be a better idea to just call `initDomain` in an init function for the test file so that any individual test that is ran will not have this issue and when running the entire test file, we should ignore the InitDomain test since it it effectively being tested via other tests

File: wrappers/golang/curves/bls12377/ntt_test.go (hunk 1 above)
Comment (jeremyfelder) [reply wrappers/golang/curves/bls12377/ntt_test.go:123]: > Consistently on running only this test

This makes sense. All the other tests as well that rely on initDomain will have the same issue.

> randomly on CI/running all the tests

I guess the `go test` engine doesn't have a deterministic order for calling tests... i'll look into it but I think my suggestion [here](https://github.com/ingonyama-zk/icicle/pull/461#pullrequestreview-1984884792) is likely the best solution for now

File: wrappers/golang/curves/bls12377/ntt_test.go [hunk 2]
```diff
@@ -72,11 +77,6 @@ func TestNTTGetDefaultConfig(t *testing.T) {
 	assert.ElementsMatch(t, cosetGenField.GetLimbs(), actual.CosetGen)
 }
 
-func TestInitDomain(t *testing.T) {
-	cfg := GetDefaultNttConfig()
-	assert.NotPanics(t, func() { initDomain(largestTestSize, cfg) })
-}
-
```
[the remark this answers is not in the document]
Comment (LeonHibnik) [reply wrappers/golang/curves/bls12377/ntt_test.go]: Good idea

**Review by jeremyfelder — APPROVED** [review]

**Pull request merged**